tests/helpers/globify.ts

```diff
@@ -0,0 +1,7 @@
+export type Globify<S extends string> = S extends `${infer Head}_${infer Tail}`
+  ? '*' | `${Head}*` | `${Head}_${Globify<Tail>}`
+  : S | '*'
+
+type Test = Globify<'foo_bar' | 'foo_bar_baz' | 'yeet'>
```

Comment:
Do we need these?

Reply:
not neccessarily - (I used them for verifying that the type works as expected) - but they don't really hurt either, as they'll never be shipped and are used only be the editor

Reply:
Sure, but why keep them if they're not doing anything?

Reply:
good point. I've transformed them into type-level tests: https://github.com/netlify/zip-it-and-ship-it/pull/702/files/7f867da77363718059c2d449e91b8750f9926b86..3ef4abe7fa6e91e9f1657b4e18187ade118971f5

so now they work as documentation & tests :)
(since we haven't integrated our test suite with TypeScript yet, this won't do anything apart from being squiggly in the editor - but that's a start ^^)